### PR TITLE
Fix regression in regenerating protobuf source

### PR DESCRIPTION
**Pull request by andygrove**
## Which issue does this PR close?

Closes https://github.com/apache/arrow-datafusion/issues/8602

## Rationale for this change

Fix regression in generating protobuf source with protobuf 3.12, which is the recommended version in the documentation.

## What changes are included in this PR?

Remove redundant `optional` keyword.

## Are these changes tested?

Existing tests.

### CODE REVIEW

**Comment by andygrove** [comment]
@alamb The use of the `optional` keyword seems nicer, but this was the only place that we were using it, so it seems better to remove it and make it consistent with the rest of the file for now? The alternative is updating docs to require the use of a newer version of protoc, but I am not sure of the impact of that on downstream projects.

File: datafusion/proto/src/logical_plan/from_proto.rs [hunk 1]
```diff
@@ -1338,7 +1338,11 @@ pub fn parse_expr(
             in_list.negated,
         ))),
         ExprType::Wildcard(protobuf::Wildcard { qualifier }) => Ok(Expr::Wildcard {
-            qualifier: qualifier.clone(),
+            qualifier: if qualifier.is_empty() {
```
Comment (alamb) [inline datafusion/proto/src/logical_plan/from_proto.rs:1341]: I think this is fine -- I don't think an empty string `""` is a valid qualifier anyways so not being able to distinguish between `""` and `None` is not important

**Review by alamb — APPROVED** [review]
Thanks @andygrove

**Comment by edmondop** [comment]
> @alamb The use of the `optional` keyword seems nicer, but this was the only place that we were using it, so it seems better to remove it and make it consistent with the rest of the file for now? The alternative is updating docs to require the use of a newer version of protoc, but I am not sure of the impact of that on downstream projects.

Sorry for the late comment. I think it would be great if we align the version in the docs with the version used by CI/CD pipeline. Will submit a PR later

**Pull request merged**